unison-core/src/Unison/Type.hs

```diff
@@ -68,9 +67,10 @@ bindNames
   -> Names.Names0
   -> Type v a
   -> Names.ResolutionResult v a (Type v a)
-bindNames keepFree ns t = let
+bindNames keepFree ns0 t = let
+  ns = Names.Names ns0 mempty
   fvs = ABT.freeVarOccurrences keepFree t
-  rs = [(v, a, R.lookupDom (Name.fromVar v) (Names.types0 ns)) | (v,a) <- fvs ]
+  rs = [(v, a, Names.lookupHQType (Name.convert $ Name.fromVar v) ns) | (v,a) <- fvs ]
```

Comment:
Generally `lookupDom` on a names relation is going to be wrong and will only find exact matches. I caught a bunch of places that were doing that and relying implicitly on the names being suffixified.